ReplayKitExample/BroadcastExtension/SampleHandler.swift

```diff
@@ -0,0 +1,227 @@
+             * Describe the supported format.
+             * For this example we cheat and assume that we will be capturing the entire screen.
+             */
+            let screenSize = UIScreen.main.bounds.size
+            let format = TVIVideoFormat()
+            format.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
+            format.frameRate = UInt(SampleHandler.kDesiredFrameRate)
+            format.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
+            return [format]
+        }
+    }
+
+    override func broadcastStarted(withSetupInfo setupInfo: [String : NSObject]?) {
+
+        TwilioVideo.audioDevice = ExampleCoreAudioDevice(audioCapturer: self)
+
+        // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+        let accessToken = "TWILIO_ACCESS_TOKEN";
+
+        let localScreenTrack = TVILocalVideoTrack(capturer: self)
+        let h264VideoCodec = TVIH264Codec()
+        let localAudioTrack = TVILocalAudioTrack()
+        let connectOptions = TVIConnectOptions.init(token: accessToken) { (builder) in
+
+            // Use the local media that we prepared earlier.
+            builder.audioTracks = [localAudioTrack!]
+            builder.videoTracks = [localScreenTrack!]
+
+            // Use the preferred video codec
+            builder.preferredVideoCodecs = [h264VideoCodec] as! [TVIVideoCodec]
+
+            // The name of the Room where the Client will attempt to connect to. Please note that if you pass an empty
+            // Room `name`, the Client will create one for you. You can get the name or sid from any connected Room.
+            if #available(iOS 12.0, *) {
+                builder.roomName = "test"
+            } else {
+                builder.roomName = setupInfo?["RoomName"] as? String
+            }
+        }
+
+        // Connect to the Room using the options we provided.
+        room = TwilioVideo.connect(with: connectOptions, delegate: self)
+
+        // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+        print("broadcastStarted")
+    }
+
+    override func broadcastPaused() {
+        // User has requested to pause the broadcast. Samples will stop being delivered.
+    }
+
+    override func broadcastResumed() {
+        // User has requested to resume the broadcast. Samples delivery will resume.
+    }
+
+    override func broadcastFinished() {
+        // User has requested to finish the broadcast.
+    }
+
+    override func processSampleBuffer(_ sampleBuffer: CMSampleBuffer, with sampleBufferType: RPSampleBufferType) {
+        RPScreenRecorder.shared().isMicrophoneEnabled = true
+        switch sampleBufferType {
+        case RPSampleBufferType.video:
+            if ((captureConsumer != nil) && room?.state == .connected) {
+                processVideoSampleBuffer(sampleBuffer)
+            }
```

Comment:
I think we should consider setting an FPS target (like 30) and dropping frames to reach that target. The goal would be to keep memory usage down, especially when many frames are in flight being encoded, or sent.

Edit: I would also not check the Room's state. Just provide the samples and defer to the media engine.